File: src/main/java/com/project/pantau/dto/report/UpdateStatusRequest.java
package com.project.pantau.dto.report;

import com.project.pantau.enums.ReportStatus;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record UpdateStatusRequest(
        @NotNull(message = "To status cannot be null")
        ReportStatus toStatus,

        @Size(max = 1000, message = "Maximum note is 1000 characters")
        String note
) {
}

File: backend/internal/dto/report/update_status_request.go
package report

import "pantau/internal/enums"

type UpdateStatusRequest struct {
	ToStatus *enums.ReportStatus `json:"to_status" validate:"required,oneof=reported acknowledged in_progress resolved closed rejected"`
	Note     *string             `json:"note" validate:"omitempty,max=1000"`
}
